dlt/common/typing.py

```diff
@@ -25,6 +26,13 @@
     runtime_checkable,
     IO,
 )
+
+if sys.version_info >= (3, 10):
+    from types import UnionType
+else:
+    # it is defined as type(int | str)
```

Comment:
NP: Mabe make the comment a bit more verbose to make it clear what we're doing here: UnionType has only been introduced in Python 3.10, so we need to override it for older python version for backward compatibility

Reply:
Sure, thanks for pointing this out. It is an important thing imo to avoid confusion in the future.

Reply:
@codingcyclist please check again :)